File: examples/node/realtime-session.mjs
import { randomUUID } from "node:crypto";

import { BeatAPIClient, BeatAPIError } from "./lib/beatapi.mjs";

const client = new BeatAPIClient();

try {
  const session = await client.createRealtimeSession(
    {
      max_duration_seconds: 60,
      allowed_origins: ["https://app.example.com"],
      metadata: { example: "node" },
    },
    { idempotencyKey: randomUUID() },
  );

  console.log(JSON.stringify({
    id: session.id,
    status: session.status,
    expires_at: session.expires_at,
    client_secret_received: Boolean(session.client_secret),
  }, null, 2));
  console.log("Return client_secret only to the exact allowed browser origin; do not log it.");
} catch (error) {
  if (error instanceof BeatAPIError) {
    console.error(
      `[${error.status}] ${error.code}: ${error.message} (${error.requestId || "no request id"})`,
    );
  } else {
    console.error(error);
  }
  process.exitCode = 1;
}

File: examples/python/realtime_session.py
import json
import uuid

from beatapi import BeatAPIClient, BeatAPIError


def main() -> None:
    client = BeatAPIClient()
    session = client.create_realtime_session(
        {
            "max_duration_seconds": 60,
            "allowed_origins": ["https://app.example.com"],
            "metadata": {"example": "python"},
        },
        idempotency_key=str(uuid.uuid4()),
    )
    print(
        json.dumps(
            {
                "id": session["id"],
                "status": session["status"],
                "expires_at": session.get("expires_at"),
                "client_secret_received": bool(session.get("client_secret")),
            },
            indent=2,
        )
    )
    print(
        "Return client_secret only to the exact allowed browser origin; do not log it."
    )


if __name__ == "__main__":
    try:
        main()
    except BeatAPIError as error:
        print(
            f"[{error.status}] {error.code}: {error} "
            f"({error.request_id or 'no request id'})"
        )
        raise SystemExit(1) from error
